src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -169,73 +171,62 @@ public void reset() {
    */
   static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
       final int hi) {
+    return intValueRange(clientSetting, lo, hi, false);
+  }
+
+  /**
+   * Text field that only accepts numbers between a certain range.
+   */
+  static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
+      final int hi, final boolean allowUnset) {
     return () -> new SelectionComponent<JComponent>() {
-      String value = clientSetting.value();
-      final JTextField component = new JTextField(value, String.valueOf(hi).length());
+      private final JSpinner component = new JSpinner(new SpinnerNumberModel(
+          toValidIntValue(clientSetting.value()), lo - (allowUnset ? 1 : 0), hi, 1));
 
       @Override
       public JComponent getUiComponent() {
-        component.setToolTipText(validValueDescription());
+        return component;
+      }
 
-        SwingComponents.addTextFieldFocusLostListener(component, () -> {
-          if (isValid()) {
-            clearError();
-          } else {
-            indicateError();
-          }
-        });
+      private int toValidIntValue(final String value) {
+        return value.isEmpty() && allowUnset ? lo - 1 : Integer.parseInt(value);
+      }
 
-        return component;
+      private String toValidStringValue(final int value) {
+        return allowUnset && value == lo - 1 ? "" : String.valueOf(value);
```

Comment:
When I set the lobby port to 0 and then click **Play Online**, the following exception is raised:

```
java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.valueOf(Integer.java:766)
	at games.strategy.triplea.settings.GameSetting.intValue(GameSetting.java:33)
	at games.strategy.engine.config.client.LobbyServerPropertiesFetcher.fetchLobbyServerProperties(LobbyServerPropertiesFetcher.java:50)
	at games.strategy.engine.framework.startup.ui.MetaSetupPanel.connectToLobby(MetaSetupPanel.java:153)
	at games.strategy.engine.framework.startup.ui.MetaSetupPanel.lambda$6(MetaSetupPanel.java:137)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at org.pushingpixels.substance.internal.utils.RolloverButtonListener.mouseReleased(RolloverButtonListener.java:126)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:352)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Reply:
@ssoloff I believe this is an issue unrelated to this PR, because when both fields are empty everything works fine.
When however the host field is set and the port field isn't this issue occurs, because the code only checks if the host field is set when checking if the properties should be overriden.
Leaving this up to you to fix in a follow-up PR
See https://github.com/triplea-game/triplea/blob/fa86e2d6516812c9d13e56fda746e7bb570ad7ba/src/main/java/games/strategy/engine/config/client/LobbyServerPropertiesFetcher.java#L47-L51

Line 50 is where the exception occurs

Reply:
Thanks for digging into that.  Confirmed this occurs on `master`, and thus is unrelated to this PR.